CMakeLists.txt

```diff
@@ -18,4 +22,10 @@ ExternalProject_Add(
     CMAKE_CACHE_ARGS -DCMAKE_CXX_FLAGS:STRING="${CMAKE_CXX_FLAGS}"
     BUILD_ALWAYS 1
     INSTALL_COMMAND ""
+)
+
+add_executable(
```

Comment:
If you're going to make an executable it's got to be done as an external project also. I wouldn't worry about making one until you've got the installation working. Once you get the installation working let's merge this PR, then we can worry about adding dependencies and/or an executable.